### PR TITLE
SVCPLAN-5555: Add support for SUSE

### PR DESCRIPTION
**Pull request by billglick**
This is being tested on `control-test-opensuse155a`.

### CODE REVIEW

**Review by bsper2 — APPROVED** [review]
Looks good

File: manifests/kerberos.pp [hunk 1]
```diff
@@ -70,50 +68,46 @@
...
       file { '/root/createhostkeytab.sh':
-        ensure => present,
+        ensure => file,
         mode   => '0700',
         source => "puppet:///modules/${module_name}/root/createhostkeytab.sh",
       }
       ## THIS MIGHT NEED TO BE SMARTER TO ALLOW FOR MULTIPLE HOSTNAMES ON ONE SERVER
       exec { 'create_host_keytab':
-        path    => [ '/usr/bin', '/usr/sbin'],
+        path    => ['/usr/bin', '/usr/sbin'],
```
Comment (billglick) [inline manifests/kerberos.pp]: Need to add path for the SUSE location of the `kadmin` command.

**Comment by billglick** [comment]
I've fixed a few things since this was originally approved...

- Added a new `$config::pam_config` that optionally configures pam files, e.g. `/etc/pam.d/common-auth` on Suse
- Updated the path for the command that creates a keytab to support alternate location for `kadmin` on Suse
- Added krb5 plugins for preauth

**Review by jakerundall — APPROVED** [review]
Looks good, thanks for chasing this down!

**Pull request merged**